source/math/FloatMatrix.ooc

```diff
@@ -123,13 +128,19 @@ FloatMatrix : cover {
 	}
 	transpose: func -> This {
 		t := this take()
-		result := This new(t dimensions swap())
-		resultElements := result elements
-		thisElements := this elements
-		for (y in 0 .. t height)
-			for (x in 0 .. t width)
-				resultElements[y + x * t height] = thisElements[x + y * t width]
-		this free(Owner Receiver)
+		result: This
+		if (t isVector && this _elements owner == Owner Receiver) {
+			result = this
+			result _dimensions = result _dimensions swap()
+		} else {
+			result = This new(t dimensions swap())
+			resultElements := result elements
+			thisElements := this elements
```

Comment:
It does not. The `elements` property does not call `free` like the other properties. This is because it returns a pointer to the object, so it would not make much sense to free it immediately after. We may want to change this in the future though, for consistency with the other properties. But as of now, this is how it works.